examples/direct_fidelity_estimation.py

```diff
@@ -29,17 +29,16 @@ def compute_characteristic_function(circuit, P_i, qubits, noise):
   n = len(P_i)
   d = 2**n
 
-  trace = 0  # rho_i / sigma_i in https://arxiv.org/pdf/1104.3835.pdf
-  for x in range(d):
-    xbin = numpy.binary_repr(x, width=n)
+  simulator = cirq.DensityMatrixSimulator()
+  density_matrix = 0  #  rho or sigma in https://arxiv.org/pdf/1104.3835.pdf
+  for step in simulator.simulate_moment_steps(circuit):
+    density_matrix = step.density_matrix()
```

Comment:
You can do this with `simulator.simulate` without the for loop

Reply:
Done.